contracts/defi/provider/CompoundV2Provider.sol

```diff
@@ -0,0 +1,405 @@
+        address[] calldata _oracles
+    ) 
+        external 
+    {
+        require(_oracles.length == 2, "CompoundV2: invalid oracles length");
+        address cToken = CompoundRegistry(_oracles[1]).getCToken(_collateral);
+        redeemUnderlying(_wallet, cToken, _collateralAmount);
+        address comptroller = _oracles[0];
+        exitMarketIfNeeded(_wallet, cToken, comptroller);
+    }
+
+    /**
+     * @dev Increases the debt by borrowing more token from a loan identified by its ID.
+     * @param _wallet The target wallet.
+     * @param _loanId The ID of the loan if any, 0 otherwise.
+     * @param _debtToken The token borrowed.
+     * @param _debtAmount The amount of token to borrow.
+     * @param _oracles (optional) The address of one or more oracles contracts that may be used by the provider to query information on-chain.
+     */
+    function addDebt(
+        BaseWallet _wallet, 
+        bytes32 _loanId, 
+        address _debtToken, 
+        uint256 _debtAmount, 
+        address[] calldata _oracles
+    ) 
+        external 
+    {
+        require(_oracles.length == 2, "CompoundV2: invalid oracles length");
+        address dToken = CompoundRegistry(_oracles[1]).getCToken(_debtToken);
+        address comptroller = _oracles[0];
+        enterMarketIfNeeded(_wallet, dToken, comptroller);
+        borrow(_wallet, dToken, _debtAmount);
+    }
+
+    /**
+     * @dev Decreases the debt by repaying some token from a loan identified by its ID.
+     * @param _wallet The target wallet.
+     * @param _loanId The ID of the loan if any, 0 otherwise.
+     * @param _debtToken The token to repay.
+     * @param _debtAmount The amount of token to repay.
+     * @param _oracles (optional) The address of one or more oracles contracts that may be used by the provider to query information on-chain.
+     */
+    function removeDebt(
+        BaseWallet _wallet, 
+        bytes32 _loanId, 
+        address _debtToken, 
+        uint256 _debtAmount, 
+        address[] calldata _oracles
+    ) 
+        external
+    {
+        require(_oracles.length == 2, "CompoundV2: invalid oracles length");
+        address dToken = CompoundRegistry(_oracles[1]).getCToken(_debtToken);
+        repayBorrow(_wallet, dToken, _debtAmount);
+        address comptroller = _oracles[0];
+        exitMarketIfNeeded(_wallet, dToken, comptroller);
+    }
+
+    /**
+     * @dev Gets information about a loan identified by its ID.
+     * @param _wallet The target wallet.
+     * @param _loanId The ID of the loan if any, 0 otherwise.
+     * @param _oracles (optional) The address of one or more oracles contracts that may be used by the provider to query information on-chain.
+     * @return a status [0: no loan, 1: loan is safe, 2: loan is unsafe and can be liquidated] and the estimated ETH value of the loan
+     * combining all collaterals and all debts. When status = 1 it represents the value that could still be borrowed, while with status = 2
```

Comment:
"and the estimated ETH value of the loan combining all collaterals and all debts" -> "and a value (in ETH) representing the value that could still be borrowed when status = 1; or the value of the collateral that should be added to avoid liquidation when status = 2"

Reply:
Done.